projects/s-rxjs-utils/src/lib/map-and-cache-elements.ts

```diff
@@ -4,11 +4,13 @@ import { map } from "rxjs/operators";
 /**
  * Applies a given function to each item in the upstream array and emits the result. Each item is then cached using the key generated by `buildCacheKey` so that the next emission contains references to the matching objects from previous emission, without running `buildDownstreamItem` again. The cache is only held between successive emissions.
```

Comment:
`[yellow]`
There was a sneaky comment hiding under my suggested rewording here from the last round of review:
> It would also be nice to also include a sentence "This is useful e.g. when ..."